sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ImageAnalysisSkill.java

```diff
@@ -1,28 +1,28 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
 
 package com.azure.search.documents.indexes.models;
 
 import com.azure.core.annotation.Fluent;
+import com.azure.core.annotation.JsonFlatten;
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonSetter;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import com.fasterxml.jackson.annotation.JsonTypeName;
-
-import java.util.Arrays;
 import java.util.List;
 
-/**
- * A skill that analyzes image files. It extracts a rich set of visual features
- * based on the image content.
- */
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type")
+/** A skill that analyzes image files. It extracts a rich set of visual features based on the image content. */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata\\.type")
```

Comment:
is "@odata\\.type" intented

Reply:
Unfortunately, this is a case where code generation is a little over eager on using `JsonFlatten`. When `JsonFlatten` is applied to a class all `JsonProperty` identifiers using a `.` will be treated as getter like calls, so `odata.type` would expected the following JSON:

```json
"odata": {
    "type": "somvalue"
}
```

But in reality the JSON is being returned as `"odata.type": "somevalue"`.